File: server/router/order/order.go
package order

import (
	v1 "gin-vue-admin/api/v1"
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

type OrderRouter struct{}

func (s *OrderRouter) InitOrderRouter(Router *gin.RouterGroup) {
	orderRouter := Router.Group("order").Use(middleware.OperationRecord())
	orderWithNotRecorderRouter := Router.Group("order")
	orderApi := v1.ApiGroupApp.OrderApiGroup.OrderApi
	{
		orderRouter.POST("invoice/apply", orderApi.ApplyInvoice) // 申请发票
	}
	{
		orderWithNotRecorderRouter.GET("overview", orderApi.GetOrderOverview)            // 获取财务总览
		orderWithNotRecorderRouter.POST("usage/list", orderApi.GetUsageList)             // 使用详情列表
		orderWithNotRecorderRouter.POST("transaction/list", orderApi.GetTransactionList) // 收支明细列表
		orderWithNotRecorderRouter.POST("order/list", orderApi.GetOrderList)             // 订单列表
		orderWithNotRecorderRouter.POST("invoice/list", orderApi.GetInvoiceList)         // 发票记录列表
	}
}

File: web/src/api/order.ts
import service from '@/utils/request'

// 获取财务概览
export const getOrderOverview = (params?: any) => {
    return service({
        url: '/api/v1/order/overview',
        method: 'get',
        params
    })
}

// 获取使用详情列表
export const getUsageList = (data?: any) => {
    return service({
        url: '/api/v1/order/usage/list',
        method: 'post',
        data
    })
}

// 获取收支明细列表
export const getTransactionList = (data?: any) => {
    return service({
        url: '/api/v1/order/transaction/list',
        method: 'post',
        data
    })
}

// 获取订单列表
export const getOrderList = (data?: any) => {
    return service({
        url: '/api/v1/order/order/list',
        method: 'post',
        data
    })
}

// 获取发票记录
export const getInvoiceList = (data?: any) => {
    return service({
        url: '/api/v1/order/invoice/list',
        method: 'post',
        data
    })
}

// 申请发票
export const applyInvoice = (data?: any) => {
    return service({
        url: '/api/v1/order/invoice/apply',
        method: 'post',
        data
    })
}
